### PR TITLE
Output tedge config control messages to stderr

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

Move `tedge config` user directed messages to stderr so actual data output and user-directed messages can be more easily told apart and parsed as described in #1721.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge/tests/main.rs [hunk 1]
```diff
@@ -88,7 +88,7 @@ mod tests {
         get_device_id_cmd
             .assert()
             .success()
-            .stdout(predicate::str::contains("device.id"));
+            .stderr(predicate::str::contains("device.id"));
```
[the remark this answers is not in the document]
Comment (Bravo555) [reply crates/core/tedge/tests/main.rs]: Looks like it's actually another error message:

```


failures:

---- tests::run_create_certificate stdout ----
thread 'tests::run_create_certificate' panicked at 'Unexpected stderr, failed var.contains('device.id' is not set)
├── var: The provided config key: 'device.id' is not configurable
└── var as str: The provided config key: 'device.id' is not configurable
```

[the remark this answers is not in the document]
Comment (didier-wenzek) [reply crates/core/tedge/tests/main.rs]: Oh true. This is read from the certificate. So let's check against `is not configurable`.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Comment by reubenmiller** [comment]
@Bravo555 @didier-wenzek I have to updated some of the integration test bootstrap script as it is failing now due to this change (that is why the integration tests are failing). I'm working on the changes and reach out to @Bravo555

**Pull request merged**